servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyIoExecutors.java

```diff
@@ -39,35 +40,92 @@ private NettyIoExecutors() {
         // No instances.
     }
 
+    /**
+     * Create a new {@link NettyIoExecutor} with the default number of {@code ioThreads}.
+     *
+     * @param <T> Type of the IO thread instances created by factory.
+     * @param threadFactory the {@link io.servicetalk.transport.api.IoThreadFactory} to use. If possible you should use
+     * an instance of {@link IoThreadFactory} as it allows internal optimizations.
+     * @return The created {@link IoExecutor}
+     */
+    public static <T extends Thread & IoThread> EventLoopAwareNettyIoExecutor createIoExecutor(
+            io.servicetalk.transport.api.IoThreadFactory threadFactory) {
+        return createIoExecutor(getRuntime().availableProcessors() * 2, threadFactory);
+    }
+
     /**
      * Create a new {@link NettyIoExecutor} with the default number of {@code ioThreads}.
      *
      * @param threadFactory the {@link ThreadFactory} to use.
      * @return The created {@link IoExecutor}
+     * @deprecated Future versions of ServiceTalk will require a
+     * {@link io.servicetalk.transport.api.IoThreadFactory} for creating {@link IoExecutor} threads, use
+     * {@link #createEventLoopGroup(int, io.servicetalk.transport.api.IoThreadFactory)} instead.
```

Comment:
I've prepared a patch instead of comments for this class:
```patch
Index: servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyIoExecutors.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyIoExecutors.java b/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyIoExecutors.java
--- a/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyIoExecutors.java	(revision c8bad3aac7ec362fb97e07e84e1f0e8f2633162b)
+++ b/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyIoExecutors.java	(date 1629230887240)
@@ -16,6 +16,7 @@
 package io.servicetalk.transport.netty.internal;
 
 import io.servicetalk.transport.api.IoExecutor;
+import io.servicetalk.transport.api.IoThreadFactory;
 import io.servicetalk.transport.api.IoThreadFactory.IoThread;
 
 import io.netty.channel.EventLoop;
@@ -44,23 +45,22 @@
      * Create a new {@link NettyIoExecutor} with the default number of {@code ioThreads}.
      *
      * @param <T> Type of the IO thread instances created by factory.
-     * @param threadFactory the {@link io.servicetalk.transport.api.IoThreadFactory} to use. If possible you should use
-     * an instance of {@link IoThreadFactory} as it allows internal optimizations.
+     * @param threadFactory the {@link IoThreadFactory} to use. If possible you should use an instance of
+     * {@link NettyIoThreadFactory} as it allows internal optimizations.
      * @return The created {@link IoExecutor}
      */
     public static <T extends Thread & IoThread> EventLoopAwareNettyIoExecutor createIoExecutor(
-            io.servicetalk.transport.api.IoThreadFactory threadFactory) {
+            IoThreadFactory<T> threadFactory) {
         return createIoExecutor(getRuntime().availableProcessors() * 2, threadFactory);
     }
 
     /**
      * Create a new {@link NettyIoExecutor} with the default number of {@code ioThreads}.
      *
-     * @param threadFactory the {@link ThreadFactory} to use.
+     * @param threadFactory the {@link ThreadFactory} to use. If possible you should use an instance of
+     * {@link NettyIoThreadFactory} as it allows internal optimizations.
      * @return The created {@link IoExecutor}
-     * @deprecated Future versions of ServiceTalk will require a
-     * {@link io.servicetalk.transport.api.IoThreadFactory} for creating {@link IoExecutor} threads, use
-     * {@link #createEventLoopGroup(int, io.servicetalk.transport.api.IoThreadFactory)} instead.
+     * @deprecated Use {@link #createIoExecutor(IoThreadFactory)}.
      */
     @Deprecated
     public static EventLoopAwareNettyIoExecutor createIoExecutor(ThreadFactory threadFactory) {
@@ -72,12 +72,12 @@
      *
      * @param <T> Type of the IO thread instances created by factory.
      * @param ioThreads number of threads.
-     * @param threadFactory the {@link io.servicetalk.transport.api.IoThreadFactory} to use. If possible you should use
-     * an instance of {@link IoThreadFactory} as it allows internal optimizations.
+     * @param threadFactory the {@link IoThreadFactory} to use. If possible you should use an instance of
+     * {@link NettyIoThreadFactory} as it allows internal optimizations.
      * @return The created {@link IoExecutor}
      */
     public static <T extends Thread & IoThread> EventLoopAwareNettyIoExecutor createIoExecutor(
-            int ioThreads, io.servicetalk.transport.api.IoThreadFactory threadFactory) {
+            int ioThreads, IoThreadFactory<T> threadFactory) {
         validateIoThreads(ioThreads);
         return new EventLoopGroupIoExecutor(createEventLoopGroup(ioThreads, threadFactory), true);
     }
@@ -87,11 +87,9 @@
      *
      * @param ioThreads number of threads.
      * @param threadFactory the {@link ThreadFactory} to use. If possible you should use an instance of
-     * {@link IoThreadFactory} as it allows internal optimizations.
+     * {@link NettyIoThreadFactory} as it allows internal optimizations.
      * @return The created {@link IoExecutor}
-     * @deprecated Future versions of ServiceTalk will require a {@link io.servicetalk.transport.api.IoThreadFactory}
-     * for creating {@link IoExecutor} threads, use
-     * {@link #createIoExecutor(int, io.servicetalk.transport.api.IoThreadFactory)} instead.
+     * @deprecated Use {@link #createIoExecutor(int, IoThreadFactory)}.
      */
     @Deprecated
     public static EventLoopAwareNettyIoExecutor createIoExecutor(int ioThreads, ThreadFactory threadFactory) {
@@ -99,16 +97,8 @@
         return new EventLoopGroupIoExecutor(createEventLoopGroup(ioThreads, threadFactory), true);
     }
 
-    /**
-     * Create a new {@link EventLoopGroup}.
-     *
-     * @param <T> Type of the IO thread instances created by factory.
-     * @param ioThreads number of threads
-     * @param threadFactory the {@link io.servicetalk.transport.api.IoThreadFactory} to use.
-     * @return The created {@link IoExecutor}
-     */
-    public static <T extends Thread & IoThread> EventLoopGroup createEventLoopGroup(int ioThreads,
-            io.servicetalk.transport.api.IoThreadFactory threadFactory) {
+    private static <T extends Thread & IoThread> EventLoopGroup createEventLoopGroup(int ioThreads,
+            IoThreadFactory<T> threadFactory) {
         validateIoThreads(ioThreads);
         return isEpollAvailable() ? new EpollEventLoopGroup(ioThreads, threadFactory) :
                 isKQueueAvailable() ? new KQueueEventLoopGroup(ioThreads, threadFactory) :
@@ -118,12 +108,12 @@
     /**
      * Create a new {@link EventLoopGroup}.
      *
-     * @param ioThreads number of threads
+     * @param ioThreads number of threads.
      * @param threadFactory the {@link ThreadFactory} to use.
      * @return The created {@link IoExecutor}
-     * @deprecated Future versions of ServiceTalk will require a
-     * {@link io.servicetalk.transport.api.IoThreadFactory} for creating {@link IoExecutor} threads, use
-     * {@link #createEventLoopGroup(int, io.servicetalk.transport.api.IoThreadFactory)} instead.
+     * @deprecated Use {@link EventLoopAwareNettyIoExecutors#toEventLoopAwareNettyIoExecutor(IoExecutor)} and
+     * {@link EventLoopAwareNettyIoExecutor#eventLoopGroup()} with any {@link NettyIoExecutor} created by other factory
+     * methods in this class.
      */
     @Deprecated
     public static EventLoopGroup createEventLoopGroup(int ioThreads, ThreadFactory threadFactory) {

```

Reply:
`pbpaste | git apply` works great for these patches.